### PR TITLE
ENH: Expose coord_transform to end-users

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
And improve its docstring to make it easier to use for end-users

### CODE REVIEW

File: nilearn/image/resampling.py [hunk 1]
```diff
@@ -103,25 +103,42 @@ def coord_transform(x, y, z, affine):
 
         Returns
         -------
-        x : number or ndarray
+        x : number or ndarray (same shape as input)
             The x coordinates in the output space
-        y : number or ndarray
+        y : number or ndarra (same shape as input)y
```
Comment (eickenberg) [inline nilearn/image/resampling.py]: the y somehow went to the end of the line :)

File: nilearn/image/resampling.py [hunk 2]
```diff
@@ -103,25 +103,42 @@ def coord_transform(x, y, z, affine):
...
             The y coordinates in the output space
-        z : number or ndarray
+        z : number or ndarra (same shape as input)y
```
Comment (eickenberg) [inline nilearn/image/resampling.py]: same here, ndarray got split

**Comment by GaelVaroquaux** [comment]
> as visible in 126 in the doctest, when just numbers are input (one
> coordinate), this returns singleton/0-dimensional arrays. Is this
> wanted? Is there a policy on these? They have always seemed a bit weird
> to me. (Calling ndarray.item on them would turn them back to numbers)

OK, I'll fix that :). It's just that its arrays manipulation under the
hood, which accepts numbers, but is really arrays.

**Comment by bthirion** [comment]
LGTM. Will merge tomorrow unless I hear otherwise.

**Comment by KamalakerDadi** [comment]
Question: nilearn.image module mostly operated on Nifti images. So, is it the right place to import ?

**Comment by GaelVaroquaux** [comment]
> Question: nilearn.image module mostly operated on Nifti images. So, is it the
> right place to import ?

Not sure. Do you have a better suggestion. It seems not unreasonnable to
me.

**Comment by KamalakerDadi** [comment]
Not a better suggestion but a suggestion. in plotting ?

**Comment by KamalakerDadi** [comment]
Please update the import in all examples. I think it is a better way to expose.

**Comment by GaelVaroquaux** [comment]
> Not a better suggestion but a suggestion. in plotting ?

Plotting depends on matplotlib, so I would rather avoid.

File: nilearn/image/resampling.py [hunk 3]
```diff
@@ -103,25 +104,45 @@ def coord_transform(x, y, z, affine):
...
             The z coordinates in the output space
 
         Warning: The x, y and z have their Talairach ordering, not 3D
         numy image ordering.
+
+    Examples
+    --------
+    Transform data from coordinates to brain space. The "affine" matrix
+    can be found as the ".affine" attribute of a nifti image, or using
+    the "get_affine()" method for old installations::
```
Comment (KamalakerDadi) [inline nilearn/image/resampling.py]: old installations -> old version of nibabel installations.

**Comment by KamalakerDadi** [comment]
>fwiw there is another typo "numy" in 114

There is a minor comment left by @eickenberg

**Comment by KamalakerDadi** [comment]
LGTM apart my minor comments left out.

**Comment by KamalakerDadi** [comment]
>fwiw there is another typo "numy" in 114

@eickenberg do you know what it means "numy" ?

**Comment by GaelVaroquaux** [comment]
Probably "numpy". I can't check right now. 

⁣Sent from my phone, please excuse typos and briefness​

**Comment by GaelVaroquaux** [comment]
Closing as superseeded in #1538

**Pull request closed without merging**